ttm_module_publish.gradle

```diff
@@ -0,0 +1,66 @@
+enum BuildType {
+    DEVELOP("-SNAPSHOT"), BETA("-BETA"), RELEASE(""), FEATURE("-FEATURE")
+
+    BuildType(String classifier) { this.classifier = classifier }
+    private final String classifier
+
+    String toString() { return classifier }
+
+    static fromBranch(String branch) {
+        // We use git flow - http://nvie.com/posts/a-successful-git-branching-model/
+        // The defaults use release/, hotfix/, etc instead of release- and hotfix- mentioned above
+        if (branch.equals("master")) {
+            //Our only releases are on master
+            return RELEASE
+        } else if (branch.startsWith("release/") || branch.startsWith("hotfix/")) {
+            //Both releases and hotfixes are prereleases - a new version that will show up as a full release later
+            return BETA
+        } else if (branch.equals("develop")) {
+            return DEVELOP
+        } else {
+            return FEATURE
+        }
+    }
+}
+
+apply plugin: 'maven-publish'
+
+ext {
+    buildType = BuildType.fromBranch(project.findProperty('branch') ?: 'feature')
+    artifactoryUrl = 'https://artifactory.trimble.tools/artifactory'
+    publishRepositoryKey = "ttm-mobile-ecosystem-${buildType == BuildType.RELEASE ? 'release' : 'snapshot'}-maven"
+}
+
+publishing {
+    publications {
+        aar(MavenPublication) {
+            groupId 'com.trimble'
+            version = "${rootProject.version}${buildType.toString()}"
+            artifactId "${rootProject.name}-${project.getName()}"
+
+            artifact("$buildDir/outputs/aar/${project.getName()}-release.aar")
```

Comment:
Is it possible to use `android.archiveName` here somehow so we're not assuming that the default of `project.getName()` is being used?

Reply:
Sorry, I meant `archivesBaseName`. It's more likely that someone would change the `archivesBaseName` than manipulate the name somewhere else, though, right? I'm just thinking that covers most cases where changes are made. I would think that grabbing the outputs of `android.libraryVariants` probably wouldn't help if someone is making changes through a custom method somewhere. 

If someone is naming their outputs using some custom method, would we even be able to grab the right AARs by using any kind of wildcard? I mean, we don't want the debug output, only the release output, so we'd already be unsure what should be published. I think just publishing the `${android.archivesBaseName}-release.aar` is okay for now.

Reply:
Ok fully agree!